backend/telemetry_core/src/state/chain.rs

```diff
@@ -35,6 +36,298 @@ id_type! {
 pub type Label = Box<str>;
 
 const STALE_TIMEOUT: u64 = 2 * 60 * 1000; // 2 minutes
+const STATS_UPDATE_INTERVAL: Duration = Duration::from_secs(5);
```

Comment:
I haven't looked at the below code in detail, but I think that we should split it (or most of it) out into a separate file (`chain_stats.rs`?); the current file has grown quite large, and I think it's easier to read if it begins with the `Chain` struct and impl (with any other odd bits after).